### PR TITLE
Add basic documentation to the website

### PR DESCRIPTION
**Pull request by criccomini**
I'm adding some very basic architecture, configuration, introduction, and
quickstart docs.

### CODE REVIEW

File: docs/quickstart.md [hunk 1]
```diff
@@ -2,4 +2,65 @@
...
 
-# Quickstart
\ No newline at end of file
+# Quickstart
+
+SlateDB is a Rust library. It doesn't currently ship with any language bindings, so you must use Rust or generate your own bindings.
+
+## Installation
+
+SlateDB is not published to crates.io yet. Sdd the following to your `Cargo.toml`:
```
Comment (criccomini) [inline docs/quickstart.md:11]: Add

File: docs/architecture.md [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+4. When the write succeeds, the immutable WAL table is notified, which notifies all `await`'ing writers.
+5. The mutable WAL table is then inserted into the mutable memtable.
+6. When the memtable reaches a `l0_sst_size_bytes`, it is frozen and written as an L0 SSTable in the object store's `compacted` directory.
+
+## Reads
+
+SlateDB's read path is as follows:
+
+1. A `get` call is made on the client.
+2. The key is returned from the mutable memtable if found.
```
Comment (criccomini) [inline docs/architecture.md:46]: Key should be value for all of these.

File: docs/architecture.md [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+* **wal_id_last_compacted**: The last WAL ID that was contained in a memtable written to L0 at the time the manifest was written..
+* **wal_id_last_seen**: The most recent WAL ID seen at the head of the WAL at the time the manifest was written. WAL SSTables older than this ID should not be read and are eligible for garbage collection.
+* **l0_last_compacted**: The Last L0 SSTable that was compacted at the time the manifest was written. L0 SSTables older than this ID should not be read and are eligible for garbage collection.
+* **l0**: A list of currently available L0 SSTables.
+* **compacted**: A list of sorted runs (SRs) that are currently available to read.
+* **snapshots**: A list of read snapshots. This feature is not yet implemented, but will allow clients to create snapshots. Snapshots will allow writers to have multi-version concurrency control (MVCC) semantics. Readers can use snapshots to ensure they have a consistent view of the state of the database (and that garbage collectors won't delete SSTables that are still being read).
```
Comment (criccomini) [inline docs/architecture.md:69]: Should read “secondary readers”

File: docs/architecture.md [hunk 4]
```diff
@@ -0,0 +1,79 @@
...
+:::
+
+## Manifest
+
+SlateDB's manifest file contains the current state of the database, including:
+
+* **manifest_id**: An auto-incrementing ID that's incremented every time a new manifest is written.
+* **writer_epoch**: The current writer epoch. This field is used to detect zombie writers. There can be only one active writer at a time. Older writers are fenced off by the newer writer by incrementing this epoch.
+* **compactor_epoch**: The current compactor epoch. As with the `writer_epoch`, this field is used to guarantee that there is only one active compactor at a time.
+* **wal_id_last_compacted**: The last WAL ID that was contained in a memtable written to L0 at the time the manifest was written..
```
Comment (criccomini) [inline docs/architecture.md:64]: Double period

File: docs/architecture.md [hunk 5]
```diff
@@ -0,0 +1,79 @@
...
+* **Sorted runs (SRs)**: A sequence of compacted, range partitioned SSTables that are treated as a single logical table.
+
+## Writes
+
+SlateDB's write path is as follows:
+
+1. A `put` call is made on the client.
+2. The key/value pair is written to the mutable, in-memory WAL table.
+3. After `flush_ms` milliseconds, the mutable WAL table is frozen and an asynchronous write to object storage is triggered.
+4. When the write succeeds, the immutable WAL table is notified, which notifies all `await`'ing writers.
```
Comment (criccomini) [inline docs/architecture.md:37]: Awaiting clients

File: docs/architecture.md [hunk 6]
```diff
@@ -0,0 +1,79 @@
...
+5. The mutable WAL table is then inserted into the mutable memtable.
```
Comment (criccomini) [inline docs/architecture.md:38]: Immutable WAL

File: docs/architecture.md [hunk 7]
```diff
@@ -0,0 +1,79 @@
...
+* **l0_last_compacted**: The Last L0 SSTable that was compacted at the time the manifest was written. L0 SSTables older than this ID should not be read and are eligible for garbage collection.
+* **l0**: A list of currently available L0 SSTables.
+* **compacted**: A list of sorted runs (SRs) that are currently available to read.
+* **snapshots**: A list of read snapshots. This feature is not yet implemented, but will allow clients to create snapshots. Snapshots will allow writers to have multi-version concurrency control (MVCC) semantics. Readers can use snapshots to ensure they have a consistent view of the state of the database (and that garbage collectors won't delete SSTables that are still being read).
+
+## Compaction
+
+SlateDB currently implements a very simple compaction strategy. The compaction strategy checks every `poll_interval`. If there are 4 or more L0 SSTables, the compactor schedules a new compaction job, which runs immediately. The job simply compacts all L0 SSTables into a single SR and writes the SSTables to the object store's `compacted` directory as the newest SR. Upon completion, the manifest to remove the L0 SSTables and add the new SR.
```
Comment (criccomini) [inline docs/architecture.md:73]: The manifest is updated

File: docs/architecture.md [hunk 8]
```diff
@@ -0,0 +1,79 @@
...
+
+## Manifest
+
+SlateDB's manifest file contains the current state of the database, including:
+
+* **manifest_id**: An auto-incrementing ID that's incremented every time a new manifest is written.
+* **writer_epoch**: The current writer epoch. This field is used to detect zombie writers. There can be only one active writer at a time. Older writers are fenced off by the newer writer by incrementing this epoch.
+* **compactor_epoch**: The current compactor epoch. As with the `writer_epoch`, this field is used to guarantee that there is only one active compactor at a time.
+* **wal_id_last_compacted**: The last WAL ID that was contained in a memtable written to L0 at the time the manifest was written..
+* **wal_id_last_seen**: The most recent WAL ID seen at the head of the WAL at the time the manifest was written. WAL SSTables older than this ID should not be read and are eligible for garbage collection.
```
Comment (rodesai) [inline docs/architecture.md:65]: > WAL SSTables older than this ID should not be read and are eligible for garbage collection.

This is true for `wal_id_last_compacted`, not `wal_id_last_seen`.

**Pull request merged**